modules/enabling-anonymous-pulls-upgraded-cluster.adoc

```diff
@@ -0,0 +1,101 @@
+// Module included in the following assemblies:
+//
+// * registry/accessing-the-registry.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="enabling-anonymous-pulls_{context}"]
+= Enabling anonymous pulls
+
+The following procedure shows you how to check if your cluster has the proper roles and permissions for anonymous pulls and, if it does not, how to enable anonymous requests to the self access reviewer source.
+
+.Procedure 
+
+. Enter the following command to see if the `system:unauthenticated` group is included in the `subjects` parameter of your `ClusterRoleBinding` object:
++
+[source,terminal]
+----
+$ oc get clusterrolebindings self-access-reviewers -o yaml
+----
++
+.Example output
++
+[source,terminal]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  annotations:
+    rbac.authorization.kubernetes.io/autoupdate: "true"
+  creationTimestamp: "2024-01-17T20:50:41Z"
+  name: self-access-reviewers
+  resourceVersion: "14477"
+  uid: d0ae964c-89ce-4b24-8905-a2318484fdd5
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: self-access-reviewer
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:authenticated
+----
+
+. If your cluster has the `self-access-reviewer` cluster role binding available to the `system:unauthenticated` group, anonymous pulls are already enabled for your cluster. You do not need to create a new cluster role binding, and can move on to the following step.
++
+If `self-access-reviewer` is not available to the `system:unauthenticated` group, or the cluster role binding does not exist, you can create a file named `add-self-access-review-unauth.yaml` and apply it with the following command:
++
+[source,yaml]
+----
+cat <<EOF | oc apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  annotations:
+    rbac.authorization.kubernetes.io/autoupdate: "true"
+  name: restore-self-access-reviewers-unauth
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: self-access-reviewer
+subjects:
+- apiGroup: rbac.authorization.k8s.io
+  kind: Group
+  name: system:unauthenticated
+EOF
+----
++
+.Example output
++
+[source,terminal]
+----
+clusterrolebinding.rbac.authorization.k8s.io/restore-self-access-reviewers-unauth created
+----
+
+. Enter the following command to create the new cluster role binding that enables anonymous requests to the self access reviewer source:
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-group system:image-puller system:unauthenticated --namespace <namespace>
+----
+
+.Verification
```

Comment:
QE uses `oc image info` to verify that anonymous pulls are enabled. What do you think of doing the same here?
You will find an example in this bug: https://issues.redhat.com/browse/OCPBUGS-33453